airbyte-integrations/connectors/source-file/integration_tests/integration_source_test.py

```diff
@@ -23,173 +23,126 @@
+    client = Client(dataset_name="output", format=file_format, url=file_path, provider=provider_config(provider_name))
+    streams = list(client.streams)
+    assert len(streams) == 1
+    assert streams[0].json_schema["properties"] == {
+        "header1": {"type": "string"},
+        "header2": {"type": "number"},
+        "header3": {"type": "number"},
+    }
+
+
+@pytest.mark.parametrize(
+    "storage_provider, url, columns_nb, config_index",
+    [
+        # epidemiology csv
+        ("HTTPS", "https://storage.googleapis.com/covid19-open-data/v2/latest/epidemiology.csv", 10, 0),
+        ("HTTPS", "storage.googleapis.com/covid19-open-data/v2/latest/epidemiology.csv", 10, 0),
+        ("local", "injected by tests", 10, 0),
+        # landsat compressed csv
+        ("GCS", "gs://gcp-public-data-landsat/index.csv.gz", 18, 1),
+        ("GCS", "gs://gcp-public-data-landsat/index.csv.gz", 18, 0),
+        # GDELT csv
+        ("S3", "s3://gdelt-open-data/events/20190914.export.csv", 58, 2),
+        ("S3", "s3://gdelt-open-data/events/20190914.export.csv", 58, 2),
+    ],
+)
+def test__read_from_public_provider(download_gcs_public_data, storage_provider, url, columns_nb, config_index):
+    config = get_config(config_index)
+    config["provider"]["storage"] = storage_provider
+    if storage_provider != "local":
         config["url"] = url
-        config["provider"]["user"] = user
-        config["provider"]["password"] = password
-        config["provider"]["host"] = host
-        run_load_dataframes(config, columns_nb, rows_nb)
+    else:
+        # inject temp file path that was downloaded by the test as URL
+        config["url"] = download_gcs_public_data
+    check_read(config, expected_columns=columns_nb)
 
 
-def run_load_dataframes(config, expected_columns=10, expected_rows=42):
-    df_list = SourceFile.load_dataframes(config=config, logger=AirbyteLogger(), skip_data=False)
-    assert len(df_list) == 1  # Properly load 1 DataFrame
-    df = df_list[0]
-    assert len(df.columns) == expected_columns  # DataFrame should have 10 columns
-    assert len(df.index) == expected_rows  # DataFrame should have 42 rows of data
-    return df
+def test__read_from_private_gcs(google_cloud_service_credentials, private_google_cloud_file):
+    config = get_config(0)
+    config["provider"]["storage"] = "GCS"
+    config["provider"]["service_account_json"] = json.dumps(google_cloud_service_credentials)
+    config["url"] = private_google_cloud_file
 
+    check_read(config)
+
+
+def test__read_from_private_aws(aws_credentials, private_aws_file):
+    config = get_config(0)
+    config["provider"]["storage"] = "S3"
+    config["provider"]["aws_access_key_id"] = aws_credentials["aws_access_key_id"]
+    config["provider"]["aws_secret_access_key"] = aws_credentials["aws_secret_access_key"]
+    config["url"] = private_aws_file
+    check_read(config)
+
+
+def get_config(index: int) -> Mapping[str, str]:
```

Comment:
I know this method was already here but I thought I'd point this out since it'd be a good change: It's very hard to understand the return value of this method because I need to magically know what each index corresponds to. I would prefer that we either take the necessary parameters as inputs directly or construct the config in the calling context.

Reply:
strongly agree...